cmd/celvet/main.go

```diff
@@ -80,24 +90,45 @@ func main() {
 		os.Exit(1)
 	}
 
-	limitErrors := celvet.CheckMaxLimits(structural)
-	for _, lintError := range limitErrors {
-		fmt.Fprintf(os.Stderr, "%s\n", lintError)
+	lintErrs := celvet.NewLinterErrors()
+	lintErrs.AddLinterErrors(celvet.CheckMaxLimits(structural))
+	lintErrs.AddLinterErrors(celvet.CheckExprCost(structural))
+	if useJSON {
+		err := emitJSON(lintErrs)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "error generating JSON output: %s\n", err)
+			os.Exit(1)
+		}
+	} else {
+		emitText(lintErrs, *humanReadable)
 	}
 
-	costErrors, compileErrors := celvet.CheckExprCost(structural)
-	for _, lintError := range costErrors {
-		if *humanReadable {
-			fmt.Fprintf(os.Stderr, "%s\n", lintError.HumanReadableError())
+	if lintErrs.HasErrors() {
+		os.Exit(1)
+	}
+}
+
+func emitJSON(l *celvet.LinterErrors) error {
+	encoder := json.NewEncoder(os.Stdout)
+	encoder.SetEscapeHTML(false)
+	return encoder.Encode(l)
+}
+
+func emitText(l *celvet.LinterErrors, humanReadable bool) {
+	for _, lintError := range l.LimitErrors {
+		fmt.Fprintf(os.Stdout, "%s\n", lintError)
```

Comment:
check if an error is returned in all calls to `Fprintf` instead of discarding them?

Reply:
Fixed - I've added an error check on all calls that weren't immediately followed by an `os.Exit(1)`, so now there's checks on the custom `flag.Usage` in addition to the entirety of `emitText`.